fence/blueprints/login/base.py

```
@@ -128,6 +128,14 @@ def _login(username, idp_name, email=None):
     redirect.
     """
     login_user(username, idp_name, email=email)
+    assert flask.g.user is not None
```

why is this required? How could that be None? I generally don't recommend using assertions like this in code outside of unit tests if you're not catching these exceptions and providing a useful message to the user - rather, you should raise an actual exception with a message. In this case though, I'm not convinced you need this at all. If anything, you should handle that case in the block below